compiler/rustc_mir_build/src/build/expr/as_constant.rs

```diff
@@ -110,15 +110,10 @@ fn lit_to_mir_constant<'tcx>(
     let LitToConstInput { lit, ty, neg } = lit_input;
     let trunc = |n| {
         let param_ty = ty::ParamEnv::reveal_all().and(ty);
-        let width =
-            tcx.layout_of(param_ty)
-                .map_err(|_| {
-                    LitToConstError::Reported(tcx.dcx().delayed_bug(format!(
-                        "couldn't compute width of literal: {:?}",
-                        lit_input.lit
-                    )))
-                })?
-                .size;
+        let width = tcx
+            .layout_of(param_ty)
+            .expect(&format!("couldn't compute width of literal: {:?}", lit_input.lit))
```

Comment:
now eagerly formats even in the success case, which is unnecessary work and may negatively impact perf